src/types.mligo

```diff
@@ -139,6 +139,24 @@ type delegate =
   }
 type delegates = (delegate, unit) big_map
 
+
+// Proposal Doubly Linked List
+
+type first_proposal_node = (proposal_key * (proposal_key option)) // The first key and its next key
+type last_proposal_node = (proposal_key * proposal_key) // The last key and its prev key
+
+type proposals_doubly_linked_list =
+  { first: first_proposal_node option
+  ; last: last_proposal_node option
+  ; map: (proposal_key, (proposal_key * proposal_key)) big_map
+  }
```

Comment:
@pasqu4le I've finished the ligo implementation. The main type for storing the order of proposals is now:

```ocaml
type first_proposal_node = (proposal_key * (proposal_key option)) // The first key and its next key
type last_proposal_node = (proposal_key * proposal_key) // The last key and its prev key

type proposals_doubly_linked_list =
  { first: first_proposal_node option
  ; last: last_proposal_node option
  ; map: (proposal_key, (proposal_key * proposal_key)) big_map
  }
```

Basically, `first` store the first proposal key, also its next key (can be empty). `last` store the last key and its prev key.

In between `first` and `last`, all the keys are store in `map` field containing their prev and next key.

All the helper function involving this type is store in `common/plist.mligo`.

Hope to get your first impression on this, let me know if I should rename the types and function as well. If all is fine, I will update the STM tests, and ensure the existing tests passed.